overlord/devicestate/devicemgr.go

```diff
@@ -89,6 +89,8 @@ type DeviceManager struct {
 	// save as rw vs ro, or mount/umount it fully on demand
 	saveAvailable bool
 
+	hasModeEnv bool
```

Comment:
judging from the actual predidate we build on this, it seems it would be nicer to capture whether we are core boot here (without using the model yet)